### PR TITLE
nerdctl inspect --format 

### PR DESCRIPTION
**Pull request by ningmingxiao**

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
Doesn't seem to work

```console
$ nerdctl image inspect --format '{{.RepoTags}}' alpine
{"Id":"sha256:021b3423115ff662225e83d7e2606475217de7b55fde83ce3447a54019a77aa2","RepoTags":["alpine:latest"],"RepoDigests":["alpine@sha256:eb3e4e175ba6d212ba1d6e04fc0782916c08e1c9d7b45892e9796141b1d379ae"],"Comment":"","Created":"2021-08-06 17:19:45.183996158 +0000 UTC","Author":"","Config":{"User":"","Env":["PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"],"Cmd":["/bin/sh"],"Volumes":null,"WorkingDir":"","Entrypoint":null,"Labels":null},"Architecture":"amd64","Os":"linux","RootFS":{"Type":"layers","Layers":["sha256:bc276c40b172b1c5467277d36db5308a203a48262d5f278766cf083947d05466"]},"Metadata":{"LastTagTime":"0001-01-01T00:00:00Z"}}
```

File: cmd/nerdctl/image_inspect.go [hunk 1]
```diff
@@ -95,11 +102,32 @@ func ImageInspectAction(clicontext *cli.Context) error {
...
+		b, err := json.MarshalIndent(f.entries, "", "    ")
+		if err != nil {
+			return err
+		}
+		fmt.Fprintln(clicontext.App.Writer, string(b))
+	case "raw", "table":
+		return errors.New("unsupported format: \"raw\" and \"table\"")
+	default:
+		var err error
+		tmpl, err = templates.Parse(format)
```
Comment (AkihiroSuda) [inline cmd/nerdctl/image_inspect.go:117]: tmpl is unused

File: cmd/nerdctl/image_inspect.go [hunk 2]
```diff
@@ -95,11 +102,32 @@ func ImageInspectAction(clicontext *cli.Context) error {
...
+		}
+		if tmpl != nil {
+			for _, value := range f.entries {
+				b, err := json.Marshal(value)
+				if err != nil {
+					return err
+				}
+				fmt.Fprintf(clicontext.App.Writer, string(b)+"\n")
+			}
+		}
```
Comment (AkihiroSuda) [inline cmd/nerdctl/image_inspect.go:129]: Lots of the codes are same across `container_inspect.go` and `image_inspect.go`.
You can move the common parts to `fmtutil.go`

File: cmd/nerdctl/image_inspect.go [hunk 3]
```diff
@@ -42,6 +44,11 @@ var imageInspectCommand = &cli.Command{
 			Usage: "Inspect mode, \"dockercompat\" for Docker-compatible output, \"native\" for containerd-native output",
 			Value: "dockercompat",
 		},
+		&cli.StringFlag{
+			Name:    "format",
+			Aliases: []string{"f"},
+			Usage:   "Format the output using the given Go template, e.g, '{{json .}}'",
```
Comment (AkihiroSuda) [inline cmd/nerdctl/image_inspect.go:50]: Please update `README.md` too

File: cmd/nerdctl/image_inspect.go (hunk 1 above)
Comment (ningmingxiao) [reply cmd/nerdctl/image_inspect.go:117]: hi, actually, I uesd this value(tmp1) in line 121 to check wether the value of tmp1 equal nil or not.

File: cmd/nerdctl/image_inspect.go (hunk 3 above)
Comment (ningmingxiao) [reply cmd/nerdctl/image_inspect.go:50]: ContainerInspectBashComplete and  imageInspectBashComplete func are the same , is it necessary to put them into fmtutil.go?

File: cmd/nerdctl/image_inspect.go (hunk 2 above)
Comment (ningmingxiao) [reply cmd/nerdctl/image_inspect.go:129]: ContainerInspectBashComplete and imageInspectBashComplete func are the same , are they necessary to put them into fmtutil.go?

File: cmd/nerdctl/image_inspect.go (hunk 1 above)
Comment (ningmingxiao) [reply cmd/nerdctl/image_inspect.go:117]: i know it

File: cmd/nerdctl/container_inspect.go [hunk 4]
```diff
@@ -74,12 +82,38 @@ func ContainerInspectAction(clicontext *cli.Context) error {
...
+	default:
+		var err error
+		tmpl, err = templates.Parse(format)
+		if err != nil {
+			return err
+		}
+		if tmpl != nil {
+			for _, value := range f.entries {
+				c, ok := value.(*dockercompat.Container)
+				if !ok {
```
Comment (fahedouch) [inline cmd/nerdctl/container_inspect.go:104]: may be a `continue` with logging Warn message to not block other entries

File: cmd/nerdctl/container_inspect.go [hunk 5]
```diff
@@ -74,12 +82,38 @@ func ContainerInspectAction(clicontext *cli.Context) error {
...
+					return errors.Errorf("%v trans to  Container  error", value)
```
Comment (fahedouch) [inline cmd/nerdctl/container_inspect.go:105]: may be a `continue` with logging Warn message to not block other entries

File: cmd/nerdctl/container_inspect.go [hunk 6]
```diff
@@ -17,14 +17,18 @@
 package main
 
 import (
+	"bytes"
 	"context"
 	"encoding/json"
 	"fmt"
+	"github.com/sirupsen/logrus"
```
Comment (AkihiroSuda) [inline cmd/nerdctl/container_inspect.go:24]: Please run `goimports`, then LGTM, thanks

File: README.md [hunk 7]
```diff
@@ -491,8 +491,9 @@ Usage: `nerctl inspect [OPTIONS] NAME|ID [NAME|ID...]`
 
 Flags:
 - :nerd_face: `--mode=(dockercompat|native)`: Inspection mode. "native" produces more information.
+- :whale: `--format`: Format the output using the given Go template, e.g, `{{json .}}`
 
-Unimplemented `docker inspect` flags: `--format`, `--size`, `--type`
+Unimplemented `docker inspect` flags:  `--size`, `--type`
```
Comment (AkihiroSuda) [inline README.md:497]: The doc of `nerdctl image inspect` has to be updated too

File: cmd/nerdctl/image_inspect.go (hunk 2 above)
Comment (AkihiroSuda) [reply cmd/nerdctl/image_inspect.go:137]: Can be another PR.

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**